examples/authenticate/authcode/authenticate.go

```diff
@@ -58,8 +65,9 @@ func completeAuth(w http.ResponseWriter, r *http.Request) {
 		http.NotFound(w, r)
 		log.Fatalf("State mismatch: %s != %s\n", st, state)
 	}
+
 	// use the token to get an authenticated client
-	client := auth.NewClient(tok)
+	client := spotify.New(spotify.HTTPClientOpt(auth.Client(r.Context(), tok)))
```

Comment:
@zmb3 , would be interested to hear your thoughts on the new style here, with the spotify HTTPClientOpt being combined with the http client returned by Client() ?

Reply:
It looks good from a feature perspective, but if we expect that this is going to be the most common way to initialize a client I think we need to make it simpler. It's quite a mouthful the way it reads right now.

It's tough to review in this format with so many changes - what might help me visualize the API is a few examples of how you might create a new client (ie the simplest/default way, and then more verbose options if you need to provide your own token or HTTP client).